src/Microsoft.DotNet.Cli.Utils/FrameworkDependencyFile.cs

```diff
@@ -45,6 +46,36 @@ public string GetNetStandardLibraryVersion()
                 ?.Version;
         }
 
+        public bool TryGetMostFitRuntimeIdentifier(IEnumerable<string> candidateRuntimeIdentifiers, out string mostFitRuntimeIdentifier)
+        {
+            mostFitRuntimeIdentifier = null;
+
+            if (!SupportsCurrentRuntime())
+            {
+                return false;
```

Comment:
The current logic when `!SupportsCurrentRuntime()` is to use/try the "portable" RID.

See

[https://github.com/dotnet/cli/blob/a71450a3059406d069f561d6938c6d13dc0acf76/src/dotnet/Program.cs#L295-L299](https://github.com/dotnet/cli/blob/a71450a3059406d069f561d6938c6d13dc0acf76/src/dotnet/Program.cs#L295-L299)

and

[https://github.com/dotnet/core-setup/blob/4b628f43fc5989c8a3923cb074b905dbc458f11c/src/corehost/cli/deps_format.cpp#L154-L164](https://github.com/dotnet/core-setup/blob/4b628f43fc5989c8a3923cb074b905dbc458f11c/src/corehost/cli/deps_format.cpp#L154-L164)

Basically, this means if you are on some new Linux distro that we don't have in our RID graph, we will at least try to use the `linux-arch` RID.